test/CadlRanchProjectsNonAzure.Tests/type-property-value-types.cs

```diff
@@ -0,0 +1,441 @@
+            Assert.AreEqual("hello", result.Property["k1"]);
+            Assert.AreEqual("world", result.Property["k2"]);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_DictionaryString_put() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetDictionaryStringClient().PutAsync(new DictionaryStringProperty(new Dictionary<string, string> { ["k1"] = "hello", ["k2"] = "world" }));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_Never_get() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetNeverClient().GetNeverAsync();
+            var result = response.Value;
+            Assert.NotNull(result);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_Never_put() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetNeverClient().PutAsync(new NeverProperty());
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_UnknownString_get() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetUnknownStringClient().GetUnknownStringAsync();
+            Assert.AreEqual("hello", response.Value.Property.ToObjectFromJson());
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_UnknownString_put() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetUnknownStringClient().PutAsync(new UnknownStringProperty(new BinaryData("\"hello\"")));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_UnknownInt_get() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetUnknownIntClient().GetUnknownIntAsync();
+            Assert.AreEqual(42, response.Value.Property.ToObjectFromJson());
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_UnknownInt_put() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetUnknownIntClient().PutAsync(new UnknownIntProperty(new BinaryData(42)));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_UnknownDict_get() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetUnknownDictClient().GetUnknownDictAsync();
+            var result = response.Value.Property.ToObjectFromJson() as Dictionary<string, object>;
+            Assert.IsNotNull(result);
+            Assert.AreEqual("hello", result["k1"]);
+            Assert.AreEqual(42, result["k2"]);
```

Comment:
we should also assert the count of `result` is `2`

Reply:
fixed